Code/Tools/AssetProcessor/Platform/Linux/native/FileWatcher/FileWatcher_linux.cpp

```diff
@@ -172,64 +153,63 @@ void FolderRootWatch::Stop()
     if (m_thread.joinable())
     {
         m_thread.join(); // wait for the thread to finish
-        m_thread = std::thread(); //destroy
     }
 }
 
 
-void FolderRootWatch::WatchFolderLoop()
+void FileWatcher::WatchFolderLoop()
 {
-    char eventBuffer[s_iNotifyReadBufferSize];
+    char eventBuffer[s_inotifyReadBufferSize];
     while (!m_shutdownThreadSignal)
     {
-        ssize_t bytesRead = ::read(m_platformImpl->m_iNotifyHandle, eventBuffer, s_iNotifyReadBufferSize);
+        ssize_t bytesRead = ::read(m_platformImpl->m_inotifyHandle, eventBuffer, s_inotifyReadBufferSize);
         if (bytesRead < 0)
         {
             // Break out of the loop when the notify handle was closed (outside of this thread)
             break;
         }
-        else if (bytesRead > 0)
+        if (!bytesRead)
         {
-            for (size_t index=0; index<bytesRead;)
+            continue;
+        }
+        for (size_t index=0; index<bytesRead;)
+        {
+            const auto* event = reinterpret_cast<inotify_event*>(&eventBuffer[index]);
+
+            if (event->mask & (IN_CREATE | IN_DELETE | IN_MODIFY | IN_MOVE))
             {
-                struct inotify_event *event = ( struct inotify_event * ) &eventBuffer[ index ];
+                const QString pathStr = QDir(m_platformImpl->m_handleToFolderMap[event->wd]).absoluteFilePath(event->name);
 
-                if (event->mask & (IN_CREATE | IN_DELETE | IN_MODIFY | IN_MOVE ))
+                if (event->mask & (IN_CREATE | IN_MOVED_TO))
                 {
-                    QString pathStr = QString("%1%2%3").arg(m_platformImpl->m_handleToFolderMap[event->wd], QDir::separator(), event->name);
-
-                    if (event->mask & (IN_CREATE | IN_MOVED_TO)) 
+                    if (event->mask & IN_ISDIR /*&& m_recursive*/)
```

Comment:
```/*&& m_recursive*/``` needed anymore?

Reply:
Eek, I missed this. This is fixed in 648a21a.